features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -57,188 +272,1561 @@ const useStyles = makeStyles((theme) => ({
+        setRevisionToRestore([revisionName, revisionId]);
+        setConfirmRestoreOpen(!confirmRestoreOpen);
+    };
+
+    const toggleDeployRevisionPopup = () => {
+        setOpenDeployPopup(!openDeployPopup);
+    };
+
+    const handleCloseDeployPopup = () => {
+        setOpenDeployPopup(false);
+        setExtraRevisionToDelete(null);
+    };
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleDeleteSelect = (event) => {
+        setExtraRevisionToDelete([event.target.value, event.target.name]);
+    };
+
+    const handleSelect = (event) => {
+        setRevision(event.target.value);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+        setExtraRevisionToDelete(null);
+    };
+
+    function checkIfDeletingLastRevision(list, revisionName) {
+        const splitList = revisionName.split(' ');
+        let splitList1;
+        if (list[list.length - 1]) {
+            const lastRevInList = list[list.length - 1].displayName;
+            splitList1 = lastRevInList.split(' ');
+        }
+        if (parseInt(splitList[1], 0) === parseInt(splitList1[1], 0)) {
+            return splitList[1];
+        }
+        return null;
+    }
+
+    const handleChange = (event) => {
+        if (event.target.checked) {
+            setSelectedEnvironment([...SelectedEnvironment, event.target.value]);
+        } else {
+            setSelectedEnvironment(
+                SelectedEnvironment.filter((env) => env !== event.target.value),
+            );
+        }
+        if (event.target.name === 'description') {
+            setDescription(event.target.value);
+        }
+    };
+
+    /**
+     * Handles deleting a revision
+     * @param {Object} revisionId the revision Id
+     * @returns {Object} promised delete
+     */
+    function deleteRevision(revisionId, revisionName) {
+        const lastRev = checkIfDeletingLastRevision(allRevisions, revisionName);
+        const promisedDelete = restApi.deleteRevision(api.id, revisionId)
+            .then(() => {
+                Alert.info(intl.formatMessage({
+                    defaultMessage: 'Revision Deleted Successfully',
+                    id: 'Apis.Details.Environments.Environments.revision.delete.success',
+                }));
+            })
+            .catch((error) => {
+                if (error.response) {
+                    Alert.error(error.response.body.description);
+                } else {
+                    Alert.error(intl.formatMessage({
+                        defaultMessage: 'Something went wrong while deleting the revision',
+                        id: 'Apis.Details.Environments.Environments.revision.delete.error',
+                    }));
+                }
+            }).finally(() => {
+                restApi.getRevisions(api.id).then((result) => {
+                    setRevisions(result.body.list);
+                    extractLastRevisionNumber(result.body.list, lastRev);
```

Comment:
If we need to keep the last revision number, setting it should be a side effect that depends on the revisions list, There is no other way of getting a different value to the max revision number?

```
extractLastRevisionNumber
```